### DISCUSSION TITLE
editor: go to definition not working

### DISCUSSION THREAD

**Discussion by cs2018ncsa** [zed-industries/zed#20409] [Feature Requests]
not sure when it started happening but i "editor: go to definition" isn't working for me via command pallete or keymap. i've tried several repostories and same issue. is it just me? any more information i can provide to maybe help troubleshoot/fix?

**Comment by leejoramo-d51** [comment]
I have noticed the same thing, and been meaning to check on this. I thought it started when I switched from to working on a Python project to JavaScript. But going back just now to the Python project, I see the same issue. So I now think the language is unrelated.

**Reply by cs2018ncsa** [reply]
good to know its not unique to me. surprised there aren't more people chiming in.

**Reply by leejoramo** [reply]
After the weekend, the problem is gone. I haven't restarted the computer, but I think Zed updated to the current version 0.160.7 

I am running Zed on KDE/Ubuntu

**Reply by cs2018ncsa** [reply]
still not working for me. gah.

**Comment by cs2018ncsa** [comment]
Finally figured out what the issue was (for me):
I had "ruff" before "pyright" in language_servers value list in settings.json. as soon as i switched them, go to definition started working.

```
  "languages": {
    "Python": {
      "language_servers": ["pyright", "ruff"],
      "format_on_save": "on",
      "formatter": [
        {
          "code_actions": {
            "source.organizeImports.ruff": true,
            "source.fixAll.ruff": true
          }
        },
        {
          "language_server": { "name": "ruff" }
        }
      ]
    }
  },

```

**Reply by mjaliz** [reply]
Thank you sooo mucch
your help bring me back to zed again
but man when you switch from vscode to zed on the same project, it's sooooo fast

**Reply by pedrosantosbr** [reply]
Worked for me as well
Just add the pyright to the language server and you guys are good to go!

**Comment by josephjohnson7222-blip** [comment]
Thank you

**Comment by venkatesh-victor** [comment]
C++, Go to definition  not working for me.

**Reply by drsybren** [reply]
I see the same in my C++ project. "Go to Definition" seems to find the uses, and not the definition.

**Reply by MariuszDrabik** [reply]
And for python too

**Reply by drsybren** [reply]
I did some digging, and it's a two-fold issue:

1. Zed can't find the definition for some reason, and
2. if it can't find the definition, it falls back to showing references (which, because it can't find the definition, excludes the definition itself).

The latter is fortunately easy to solve with this snippet in your `settings.json`:

```json
{
  "go_to_definition_fallback": "none",
}
```

The former I solved, in my CMake project, by passing `-DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DWITH_UNITY_BUILD=OFF`. I don't know if disabling the unity build is really a necessary step; I read it [in the Blender developer docs on clangd use](https://developer.blender.org/docs/handbook/tooling/clangd/), and it seems to do the trick. That might be a [Blender](https://www.blender.org/)-specific option, though -- it's Blender's sources that I'm working on.

`-DCMAKE_EXPORT_COMPILE_COMMANDS=ON` is certainly necessary. I symlinked it from my build directory into the source directory, so that Zed/clangd can pick it up.

**Comment by JosephTLyons** [comment]
Thanks for reaching out. We want to better support our community, and the best way for us to do that is through our official Discord. The team is active there and happy to help.

We’re closing the “Help and General Q&A” category on GitHub Discussions and directing all support questions to Discord going forward.

https://zed.dev/community-links

**Comment by trevor-the-developer** [comment]
Same issue with latest version Ctrl-Click or Go to Definition not working.

**Reply by trevor-the-developer** [reply]
Working now after the latest update lets hope it stays that way great work guys thank you.
